packages/fx-core/src/plugins/resource/aad/utils/common.ts

```diff
@@ -45,24 +53,66 @@ export class Utils {
       : message;
   }
 
-  public static getConfigFileName(ctx: PluginContext, isLocalDebug: boolean): string {
-    if (isLocalDebug) {
+  public static getConfigFileName(envName?: string): string {
+    if (!envName) {
       return ConfigFilePath.LocalSettings;
     } else {
-      return ConfigFilePath.State(ctx.envInfo.envName);
+      return ConfigFilePath.State(envName);
     }
   }
 
-  public static getInputFileName(ctx: PluginContext): string {
-    return ConfigFilePath.Input(ctx.envInfo.envName);
+  public static getInputFileName(envName: string): string {
+    return ConfigFilePath.Input(envName);
   }
 
-  public static async getCurrentTenantId(ctx: PluginContext): Promise<string> {
-    const tokenObject = await ctx.graphTokenProvider?.getJsonObject();
-    const tenantId: string = (tokenObject as any).tid;
+  public static async getCurrentTenantId(graphTokenProvider?: GraphTokenProvider): Promise<string> {
+    const tokenObject = await graphTokenProvider?.getJsonObject();
+    const tenantId: string = (tokenObject as any)?.tid;
     return tenantId;
   }
 
+  public static skipCreateAadForProvision(envInfo: v3.EnvInfoV3): boolean {
+    const envConfig: EnvConfig = envInfo.config as EnvConfig;
+    const envState: v3.AADApp = envInfo.state[BuiltInResourcePluginNames.aad] as v3.AADApp;
+    const objectId = envConfig.auth?.objectId ? envConfig.auth.objectId : envState.objectId;
```

Comment:
I think we should not read state here:
`const objectId = envConfig.auth?.objectId`
If there is no value from config, we should just return false in line 97